sdk/iot/core/tests/cmocka/az_iot_sas_token_tests.c

```diff
@@ -19,7 +22,38 @@
 #define TEST_EXPIRATION_STR "1578941692"
 #define TEST_KEY_NAME "iothubowner"
 
-/*
+#ifndef NO_PRECONDITION_CHECKING
+
+// This block defines the resources needed to verify precondition checking.
+// The macro assert_precondition_checked(func) alone shall be used to assert if a precondition is verified.
+// If a precondition is not verified within a function the result will be either:
+// - A crash in the test, as the the target function will continue to execute with an invalid argument, or
+// - An assert failure indicating a precondition was not tested.
+// Notice that:
+// - If assert_precondition_checked(func) is used, the module must include <setjmp.h>;
+// - If a function has two precondition checks and both are supposed to fail on a given test, assert_precondition_checked(func)
+//   is unable to distinguish which precondition has failed first. Testing precondition checking separately is advised.
+// - Tests using assert_precondition_checked(func) must not be run in parallel (!).
```

Comment:
Fyi: you could make parallel work if you used thread local storage or a map of thread ID to jmp_buf.

Reply:
True. I'll make that a note in the documentation above.